File: 1005. Maximize Sum Of Array After K Negations/Solution.py
import heapq
from typing import List


class Solution:
    def largestSumAfterKNegations(self, A: List[int], K: int) -> int:
        heapq.heapify(A)
        for _ in range(K):
            heapq.heapreplace(A, -A[0])
        return sum(A)

File: 1005. Maximize Sum Of Array After K Negations/Solution.java
import java.util.PriorityQueue;

class Solution {
    public int largestSumAfterKNegations(int[] A, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(A.length);
        for (int x : A) {
            pq.add(x);
        }
        for (int i = 0; i < K; i++) {
            pq.add(-pq.poll());
        }
        return pq.stream().mapToInt(i -> i).sum();
    }
}
